### PR TITLE
drainer: fix kafka message limit problem

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
When we use kafka as downstream, the transaction size exceed 1G can't to transport to downstream.

### What is changed and how it works?
1. Set grpc message limit to IntMax instead of `max.Int32`.
2. Add configuration `max-message-size` in drainer's config.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported variable/fields change

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note
 
### Release note

- Set grpc message limit to IntMax instead of `max.Int32`. Add configuration `max-message-size` in drainer's config.

### CODE REVIEW

File: drainer/config.go [hunk 1]
```diff
@@ -439,8 +439,11 @@ func (cfg *Config) adjustConfig() error {
 	}
 
 	if cfg.SyncerCfg.DestDBType == "kafka" {
-		maxMsgSize = maxKafkaMsgSize
-
+		if cfg.SyncerCfg.To.KafkaMaxMessageSize > 0 {
+			maxMsgSize = cfg.SyncerCfg.To.KafkaMaxMessageSize
```
Comment (glorv) [inline drainer/config.go:443]: What if `KafkaMaxMessageSize` > MaxInt32 。 I think we need to do this check here

**Review by glorv — APPROVED** [review]

File: cmd/drainer/drainer.toml [hunk 2]
```diff
@@ -164,6 +164,7 @@ port = 3306
 # kafka-addrs = "127.0.0.1:9092"
 # kafka-version = "0.8.2.0"
 # kafka-max-messages = 1024
+# kafkf-max-message-size = 1073741824 # configure max kafka **client** message size
```
Comment (amyangfei) [inline cmd/drainer/drainer.toml:167]:
```suggestion
# kafka-max-message-size = 1073741824 # configure max kafka **client** message size
```

**Comment by King-Dylan** [comment]
Maybe we also need cherry pick in 4.0

**Comment by niubell** [comment]
close https://github.com/pingcap/tidb/issues/28659

**Pull request merged**